server/queries/BattleQueries.h

```diff
@@ -33,6 +35,7 @@ class CBattleDialogQuery : public CDialogQuery
 {
 public:
 	CBattleDialogQuery(CGameHandler * owner, const BattleInfo * Bi);
+	CBattleDialogQuery(CGameHandler * owner, const BattleInfo * Bi, const SideInBattle & sideToAdd);
```

Comment:
1) Delete 1st constructor - I think it is (no longer used anywhere)
2) Use PlayerColor instead of SideInBattle - this is internal class of battle system, no need to expose it.
```suggestion
	CBattleDialogQuery(CGameHandler * owner, const BattleInfo * Bi, const PlayerColor & playerToAdd);
```